inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -814,24 +814,14 @@ void convertFunctionToICNNNetwork(const std::shared_ptr<const ::ngraph::Function
             std::string outName = layer->get_friendly_name();
             if (layer->get_output_size() != 1) outName += "." + std::to_string(i);
             DataPtr &ptr = cnnNetworkImpl->getData(outName.c_str());
-            SizeVector dims;
-            dims = layer->get_output_shape(i);
+            SizeVector dims = layer->get_output_shape(i);
             for (const auto &dim : dims) {
                 if (!dim)
                     THROW_IE_EXCEPTION << cnnLayer->type << " layer " << cnnLayer->name
                         << " has incorrect dimensions in the output data " << i;
             }
             if (!ptr && nGraphImpl && nGraphImpl->_data.find(outName) != nGraphImpl->_data.end()) {
                 ptr = nGraphImpl->_data.at(outName);
-                if (auto nData = std::dynamic_pointer_cast<InferenceEngine::details::NGraphData>(ptr)) {
-                    const auto layout =
-                        dims.size() == nData->getTensorDesc().getDims().size() ?
-                        nData->getTensorDesc().getLayout() :
-                        TensorDesc::getLayoutByDims(dims);
-
-                    nData->reset();
-                    nData->reshape(dims, layout);
-                }
```

Comment:
@ilyachur why this block of code was needed? I wonder how layout could be changed and why reshape is needed

Reply:
This code was needed for avoid situation when shapes in nGraph function were changed after data creation. As I remember we covered this situation by some tests.

Reply:
I keep this code, but without condition:
```c++
if (auto nData = std::dynamic_pointer_cast<InferenceEngine::details::NGraphData>(ptr))
```

Reply:
Ok, it looks good